File: Server/src/main/java/com/example/student/services/StudentService.java
package com.example.student.services;

import com.example.student.model.Student;

import java.util.List;

public interface StudentService {
    Student createStudent(Student student);

    List<Student> getAllStudents();

    boolean deleteStudent(Long id);

    Student getStudentById(Long id);

    Student updateStudent(Long id, Student student);

}

File: client/src/services/StudentService.js
import axios from "axios";

const STUDENT_API_BASE_URL = "http://localhost:8080/api/v1/student";

class StudentService {
  saveStudent(student) {
    return axios.post(STUDENT_API_BASE_URL, student);
  }

  getStudents() {
    return axios.get(STUDENT_API_BASE_URL);
  }

  deleteStudent(id) {
    return axios.delete(STUDENT_API_BASE_URL + "/" + id);
  }

  getStudentById(id) {
    return axios.get(STUDENT_API_BASE_URL + "/" + id);
  }

  updateStudent(student, id) {
    return axios.put(STUDENT_API_BASE_URL + "/" + id, student);
  }
}

export default new StudentService();
